src/tools/recipe2plan.ts

```diff
@@ -0,0 +1,162 @@
+import {TestVolatileMemoryProvider} from '../runtime/testing/test-volatile-memory-provider.js';
+import {RamDiskStorageDriverProvider, RamDiskStorageKey} from '../runtime/storageNG/drivers/ramdisk.js';
+import {Capabilities} from '../runtime/capabilities.js';
+import {ramDiskStorageKeyPrefixForTest, storageKeyPrefixForTest} from '../runtime/testing/handle-for-test.js';
+
+
+/**
+ * Generates Kotlin Plans from recipes in an arcs manifest.
+ *
+ * @param path path/to/manifest.arcs
+ * @return Generated Kotlin code.
+ */
+export async function recipe2plan(path: string): Promise<string> {
+  const manifest = await Runtime.parseFile(path);
+
+  const recipes = new StorageKeyRecipeResolver(manifest).resolve();
+
+  const plans = await generatePlans(recipes);
+
+  return plans.join('\n');
+}
+
+
+/**
+ * Converts each resolved recipes into a Kotlin Plan class.
+ *
+ * @param resolutions A series of resolved recipes.
+ * @return List of generated Kotlin plans
+ */
+async function generatePlans(resolutions: AsyncIterator<Recipe>): Promise<string[]> {
+  // TODO Implement
+  return [''];
+}
+
+
+/**
+ * Responsible for resolving recipes with storage keys.
+ */
+export class StorageKeyRecipeResolver {
+  private readonly runtime: Runtime;
+
+  constructor(context: Manifest) {
+    const loader = new Loader();
+    this.runtime = new Runtime({loader, context});
+  }
+
+  /**
+   * Produces resolved recipes with storage keys.
+   *
+   * TODO(alxr): Apply to long-running recipes appropriately.
+   * @throws Error if recipe fails to resolve on first or second pass.
+   * @yields Resolved recipes with storage keys
+   */
+  async* resolve(): AsyncIterator<Recipe> {
+    for (const r of this.runtime.context.allRecipes) {
+      const arc = this.runtime.newArc(this.getArcId(r), ramDiskStorageKeyPrefixForTest());
+      const opts = {errors: new Map<Recipe | RecipeComponent, string>()};
+      const resolved = await this.resolveOrNormalize(r, arc, opts);
+      if (!resolved) {
+        throw Error(`Recipe ${r.name} failed to resolve:\n${[...opts.errors.values()].join('\n')}`);
+      }
+      this.createStoresForCreateHandles(resolved, arc);
+      resolved.normalize();
+      if (!resolved.isResolved()) {
+        throw Error(`Recipe ${resolved.name} did not properly resolve!\n${resolved.toString({showUnresolved: true})}`);
+      }
+      yield resolved;
+    }
+  }
+
+  /**
+   * Resolves unresolved recipe or normalizes resolved recipe.
+   *
+   * @param recipe long-running or ephemeral recipe
+   * @param arc Arc is associated with input recipe
+   * @param opts contains `errors` map for reporting.
+   */
+  async resolveOrNormalize(recipe: Recipe, arc: Arc, opts?: IsValidOptions): Promise<Recipe | null> {
+    const normed = recipe.clone();
+    normed.normalize();
+    if (normed.isResolved()) return normed;
+
+    return await (new RecipeResolver(arc).resolve(recipe, opts));
+  }
+
+  /** @returns the arcId from annotations on the recipe when present. */
```

Comment:
`Returns the ...` rather than `@returns the` -- Summary fragments read nicer, and convey the same information.